### PR TITLE
changed <Menu> to use <ItemChoice>

### PR DESCRIPTION
**Pull request by wakooka**
So far there's a separator on each item, an update on the ItemChoice is coming to be able to specify which has a separator.
capture d ecran 2018-08-27 a 11 11 03

### CODE REVIEW

File: src/itemChoice/index.tsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ class ItemChoice extends PureComponent <ItemChoiceProps> {
     return (
       <Component
         role="option"
-        key={key}
+        key={id}
```
Comment (pith) [inline src/itemChoice/index.tsx]: 🤔 why do we have a key here ? IMO you should just remove it

File: src/menu/index.tsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 }
 
 const Menu = ({ className, items }: MenuProps) => (
-  <ul className={cc(className)} role="menu">
-    {items.map(item => <MenuItem {...item} key={item.id} />)}
-    <style jsx>{style}</style>
-  </ul>
+  <nav className={cc(className)} role="menu">
+    {items.map(item => (
+      <ItemChoice
```
Comment (pith) [inline src/menu/index.tsx]: Does it use an li tag by default ? If it's the case you might need to keep the ul tag

File: src/menu/story.tsx [hunk 3]
```diff
@@ -12,22 +14,57 @@ stories.addDecorator(withKnobs)
 const menuItems = [
   {
     id: 'menu-item-1',
-    label: 'Menu item 1',
-    url: '#',
+    label: 'Dashboard',
+    leftAddon: <HomeIcon />,
+    href: '#',
```
Comment (pith) [inline src/menu/story.tsx]: nit: `#item-1`

File: src/menu/index.tsx [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
-import style from './style'
+import ItemChoice, { ItemChoiceProps } from 'itemChoice'
 
 export interface MenuProps {
-  items: MenuItemProps[],
+  items: ItemChoiceProps[],
   className?: Classcat.Class,
 }
 
 const Menu = ({ className, items }: MenuProps) => (
```
Comment (pith) [inline src/menu/index.tsx]: It's true that the added value of this component is not huge

File: src/menu/index.tsx (hunk 2 above)
Comment (wakooka) [reply src/menu/index.tsx]: It's either a `<a>` if there's a `href` prop or a `<li>` otherwise

File: src/itemChoice/index.tsx (hunk 1 above)
Comment (wakooka) [reply src/itemChoice/index.tsx]: ping @Bidjii

Comment (Bidjii) [reply src/itemChoice/index.tsx]: Yes you could remove it, I talked about it with Loïck, I just have to make sure that Typescript doesn't throw us an error if we don't allow the props to be put here on the first place

File: src/menu/index.tsx (hunk 2 above)
Comment (Bidjii) [reply src/menu/index.tsx]: If there's a `href` props that is a string, you get a `<a>`, yes

File: src/itemChoice/index.tsx (hunk 1 above)
Comment (wakooka) [reply src/itemChoice/index.tsx]: Ok I removed it and checked, it's all good.

**Pull request merged**